### ISSUE TITLE
Code block no longer show syntax highlight

### ISSUE THREAD

**Issue by jovezhong** [facebook/docusaurus#9888]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

I have been using Docusaurus for 2+ years. Recently I found the code block no longer shows the syntax highlight
<img width="817" alt="image" src="https://github.com/facebook/docusaurus/assets/5076438/a6d4e046-27d8-4055-a701-ffcce9bb19ca">

The md is https://github.com/timeplus-io/docs/blob/main/docs/query-syntax.md
Github still shows this correctly.

### Steps to reproduce

1. Git clone https://github.com/timeplus-io/docs
2. yarn install
3. yarn start run

### Expected behavior

code block with proper syntax highlight

### Actual behavior

just a regular html code block, no syntax color

### Your environment

- Public source code: https://github.com/timeplus-io/docs/tree/main
- Public site URL: https://docs.timeplus.com
- Docusaurus version used: 3.1.1
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Node 18
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): Mac

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by slorber** [comment]
Documented as a v3.0 breaking change:
https://docusaurus.io/docs/migration/v3#prism-react-renderer-v20

**Comment by jovezhong** [comment]
Hi @slorber , the syntax highlight was working when I upgraded to v3 (I followed that migration guide and added the additional languages). But recently got this issue

**Comment by slorber** [comment]
So you did something that broke it probably, and need to figure out what. As you said, it works on v3 😅

**Comment by jovezhong** [comment]
Thanks for the reply. I tried to a few older version of prism and cannot get it back working. Appreciate if you can share any debug/troubleshooting tips. I don't see JS error about this, nor the build error. DOM elements seem to be okay. Not a burning issue but would like to fix it. Also the copy button was missing recently.

**Comment by slorber** [comment]
You should checkout older commits one by one until it works again to know which commit broke your site highlighting.

If the copy button doesn't show, it's probably that you did something reading to prism and our theme code block not even being used at all. Are you even seeing a non SQL code block being highlighted on any page?

Are you able to highlight SQL on a newly initialized site? If not, then it's a bug you should report. 


Your monorepo setup is weird. We usually use workspaces and do not declare dependencies at the root. It's not possible for me to spend time investigating a monorepo setup like this, because any package/dependency can cause issue with Docusaurus. Try isolating your site in a single repo with proper dependencies and see if it works better.

**Comment by jovezhong** [comment]
Thanks @slorber  for the reply. Yes, I guess the best way is to kind of git blame and try different versions to locate when this is broken. Not just SQL block, other bash/js/python code blocks are not working.

You mentioned the monorepo setup, but https://github.com/timeplus-io/docs is a repo only for the docs. I think I created it via Docusaurus CLI or template (cannot remember the details, it was 2-3 years ago)

The dependencies are only for Docusaurus.

Again, I will try 'time travel' back to the commit a few months ago, then do some kind of binary search to try to locate which commit breaking the code highlight (check out certain commit, yarn install, yarn run start)

**Comment by slorber** [comment]
Sorry I mistakenly saw a weird monorepo, that was probably another repro posted yesterday.

**Comment by jovezhong** [comment]
Hi @slorber , I just reported back my finding.

The issue was solved
* I tried kind of 'binary-search', to go back 2 months ago, see whether it worked, then 1 months ago, 2w ago, etc
* I located the commit for introducing GA4 https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-google-gtag
* But I do need to use GA4 and I found if I delete the yarn.lock and `yarn install` again to regenerate it, the problem is gone
* So I went back to my latest code, regenerated the yarn.lock (not via yarn install, delete it first) and the issue is solved.

I guess there could be some duplicated lib in the old yarn.lock and GA4 dependencies just made things worse. A clean deps is the key.

Nothing wrong with docusaurus or the docusaurus/plugin-google-gtag
